app/src/main/java/ru/uniquenature/myapplication/FragmentMoviesList.kt

```diff
@@ -26,13 +32,38 @@ class FragmentMoviesList : Fragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        initViews(view)
+        context?.let {
+            viewModel.loadingMovies(it)
+        }
+        viewModel.listMoviesState.observe(this.viewLifecycleOwner,this::setState)
+    }
+
+    private fun setState(state:ListMoviesViewModel.ViewModelListState)= when (state){
+        is ListMoviesViewModel.ViewModelListState.Success->setAdapter(state.list)
+        is ListMoviesViewModel.ViewModelListState.Error->errors(state.error)
+        is ListMoviesViewModel.ViewModelListState.Loading->loadingData()
+    }
+
+    private fun loadingData(){
+        progress?.isVisible = true
+    }
+
+    private fun setAdapter(movies:List<Movie>){
+        recycler?.adapter = MoviesAdapter(movies){item -> doClick(item)}
+        progress?.isVisible = false
+    }
+
+    @SuppressLint("ShowToast")
+    private fun errors(error:String){
```

Comment:
@sanya-arsenal Try to name the function more transparent for example `showError`